fh-messaging/lib/fh_metrics_man.js

```diff
@@ -202,19 +202,65 @@ MetricsManager.prototype.deleteMetricsData = function(topic, date, callback) {
   messageMan = new fhmsg.Messaging(msgConfig, self.mLogger);
   self.messageMans.push(messageMan);
   messageMan.database.on('tearUp', function() {
-    messageMan.database.db.dropCollection(collectionName, callback);
-    //remove all the data that is older than the x number of days from the intermediate tables
-    async.each(['transactionsdesttemp', 'transactionsgeotemp'], function(tempCol, cb) {
-      var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
-      var beforeDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToCalculate);
-      var query = {"_id.ts":{$lt: beforeDate.getTime()}};
-      messageMan.database.db.collection(tempCol, function(err, collection) {
-        if (err) {
-          return cb(err);
+
+    // Pruning for fhact
+    if (topic === 'fhact') {
+      var collection_names = [];
+      var prune_collection_names = [];
+      var filtered_collection_names = [];
+
+      // Retrieve the list of collections from the database
+      messageMan.database.db.listCollections({}).toArray( function(err, items) {
+        // get the collection names
+        for ( var i=0; i < items.length; i++ ) {
+          collection_names.push(items[i].name);
```

Comment:
Is it possible for an item to come back undefined?

Reply:
Hmm adding the following, would this catch such an error ? 

```
if (err) {
  self.mLogger.error(err);
  callback(err);
}
```

Reply:
@davidkirwan That is different than what i had raised but even more relevant. What you have suggested should definitely be added.

For my comment, if you add in something like
```
if (items[i] && items[i].name){
    collection_names.push(items[i].name);
}

Reply:
:+1: